### PR TITLE
fix(docs): fix scim patch api docs

### PR DESCRIPTION
**Pull request by cathteng**
Fixes https://docs.sentry.io/api/scim/update-a-teams-attributes/ to the below (encloses all json blobs with "Operations": {}). From working with Postman myself and from hearing others' experiences, this needs to be added for patching to work.

I previously tried to do this in #37949 but only updated the docstring, not the actual docs.

---

Update a team's attributes with a SCIM Group PATCH Request. Valid Operations are:
* Renaming a team:
```json
{
    "Operations": {
        "op": "replace",
        "value": {
            "id": 23,
            "displayName": "newName"
        }
    }
}
```
* Adding a member to a team:
```json
{
    "Operations": {
        "op": "add",
        "path": "members",
        "value": [
            {
                "value": 23,
                "display": "testexample@example.com"
            }
        ]
    }
}
```
* Removing a member from a team:
```json
{
    "Operations": {
        "op": "remove",
        "path": "members[value eq \"23\"]"
    }
}
```
*  Replacing an entire member set of a team:
```json
{
    "Operations": {
        "op": "replace",
        "path": "members",
        "value": [
            {
                "value": 23,
                "display": "testexample2@sentry.io"
            },
            {
                "value": 24,
                "display": "testexample3@sentry.io"
            }
        ]
    }
}
```

### CODE REVIEW

**Review by AniketDas-Tekky — CHANGES_REQUESTED** [review]
We're trying to move away from the static api docs. drf spectacular picks up the first line of the docstring as the endpoint description so you only need to remove the static docs and modify the docstring a bit.

File: src/sentry/scim/endpoints/teams.py [hunk 1]
```diff
@@ -197,10 +198,11 @@ def post(self, request: Request, organization) -> Response:
         return super().post(request, organization)
 
 
+@extend_schema(tags=["SCIM"])
 @region_silo_endpoint
 class OrganizationSCIMTeamDetails(SCIMEndpoint, TeamDetailsEndpoint):
     permission_classes = (OrganizationSCIMTeamPermission,)
-    public = {"GET", "PATCH"}
+    public = {"GET", "PATCH", "DELETE"}
```
Comment (AniketDas-Tekky) [inline src/sentry/scim/endpoints/teams.py:205]: @maxiuyuan Do we want to make DELETE public?

Comment (cathyteng17) [reply src/sentry/scim/endpoints/teams.py:205]: i was wondering this too because the docs for DELETE don't generate unless DELETE is public

**Review by AniketDas-Tekky — COMMENTED** [review]
Did you do a check on the generated docs to make sure that it looks good?

File: src/sentry/scim/endpoints/teams.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply src/sentry/scim/endpoints/teams.py:205]: Yeah, we only generate docs for Public methods, but it also means we're committing to support it.

Comment (cathyteng17) [reply src/sentry/scim/endpoints/teams.py:205]: DELETE has public docs but those were generated from the static docs https://docs.sentry.io/api/scim/delete-an-individual-team/

**Review by AniketDas-Tekky — APPROVED** [review]
~Waiting to make sure we're okay publishing DELETE. But otherwise LGTM~

Everything's good! :shipit:

**Pull request merged**